concepts/json/about.md

```diff
@@ -0,0 +1,441 @@
+# About
+
+## JSON
+
+[JSON][json] (JavaScript Object Notation) is a human readable data and file format commonly used to exchange data, in particular for web applications.
+As such, it is no surprise that is holds a special place in the Elm language.
+
+Elm provides the core modules [`Json.Decode`][decode] and [`Json.Encode`][encode] to parse and write JSON data.
+JSON Decoders are a way to _declare_ what type of data is expected and which data structure the data should map to.
+
+Let's import some aliases and types from the modules:
+
+```elm
+import Json.Decode as Decode exposing (Decoder, Error)
+import Json.Encode as Encode exposing (Value)
+```
+
+Using `Decode.decodeString : Decoder a -> String -> Result Error a` with a decoder and a JSON string will either succeed if the string confirms to the decoder specifications, or fail otherwise.
+There is no way of inspecting the raw JSON data when defining the decoder, which could feel unusual coming from imperative languages.
+
+## Basic data type decoders
+
+JSON has a small number of data types:
+
+- String: encoded with double quotes `"hello world"`
+- Boolean: either `true` or `false`
+- Number: such as `34` or `0.12`
+- Null: the absence of data `null`
+- Array: a collection of other data types `[0, null, "none", []]`
+- Object: a collection of key-value pairs `{"id": 1, "is_admin": false, "more_info": {}}`
+
+Each of these is associated with a `Decode` function:
+
+```elm
+Decode.decodeString Decode.string "\"hello\""
+    --> Ok "hello"
+
+Decode.decodeString Decode.string "null"
+    --> Err ...
+
+Decode.decodeString Decode.bool "true"
+    --> Ok True
+
+Decode.decodeString Decode.bool "0"
+    --> Err ...
+
+Decode.decodeString Decode.int "12"
+    --> Ok 12
+
+Decode.decodeString Decode.float "3.14"
+    --> Ok 3.14
+
+Decode.decodeString (Decode.null AnyValue) "null"
+    --> Ok AnyValue
+
+Decode.decodeString (Decode.list Decode.int) "[1, 2, 3]"
+    --> Ok [1, 2, 3]
+
+Decode.decodeString (Decode.list Decode.int) """[1, 2, 3, "not an int"]"""
+    --> Err ...
+
+Decode.decodeString (Decode.dict Decode.int) """{ "key1": 17, "key2": 71 }"""
+    --> Ok (Dict.fromList [("key1", 17), ("key2", 71)])
+
+Decode.decodeString (Decode.dict Decode.int) """{ "key1": 17, "key2": "seventy-one" }"""
+    --> Err ...
+```
+
+Note that `Decode.list : Decoder a -> Decoder (List a)` and `Decode.dict : Decoder a -> Decoder (Dict String a)` each expect a decoder as argument, which is expected to decode every single element elements of the data structure.
```

Comment:
I split the section in two (core data types / collections) to facilitate that.
What would you talk about specifically?
(sorry for the 3 month response time, I finally have some time to focus on this PR this week)

Reply:
Do you think it would be better if I straight up removed the references to to GeoJSON and simply described the different decoder functions?
One one hand, I liked the GeoJSON thing because it's real world, but on the other, you're right that it makes the document very long.
One thing I could do it keep it for the about document but remove it from the intro, and I would just introduce the different functions there.